pkg/cluster/provider/eksclusterprovider.go

```diff
@@ -569,11 +574,34 @@ func (p *EKSClusterProvider) initClusterData() error {
 		return fmt.Errorf("describing cluster '%s': %w", *clusterName, err)
 	}
 
+	uniqueAZs := map[string]struct{}{}
+	subnetResp, err := p.ec2Manager.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		SubnetIds: dcr.Cluster.ResourcesVpcConfig.SubnetIds,
+	})
+	if err != nil {
+		return fmt.Errorf("describing the subnets of cluster '%s': %w", *clusterName, err)
+	}
+	if subnetResp == nil {
+		return fmt.Errorf("describing subnets of cluster '%s' had empty response", *clusterName)
+	}
+	for _, subnet := range subnetResp.Subnets {
+		if subnet == nil || subnet.AvailabilityZone == nil {
+			continue
+		}
+		uniqueAZs[*subnet.AvailabilityZone] = struct{}{}
+	}
+
+	azs := []string{}
+	for az := range uniqueAZs {
+		azs = append(azs, az)
+	}
+
 	p.ClusterData = &EKSClusterData{
-		ClusterName: aws.StringValue(clusterName),
-		NodeRole:    nodeRole,
-		SubnetIDs:   aws.StringValueSlice(dcr.Cluster.ResourcesVpcConfig.SubnetIds),
-		Cluster:     dcr.Cluster,
+		ClusterName:       aws.StringValue(clusterName),
+		NodeRole:          nodeRole,
+		SubnetIDs:         aws.StringValueSlice(dcr.Cluster.ResourcesVpcConfig.SubnetIds),
```

Comment:
if you are passing the whole cluster through now, do you still need to pass through subnet IDs?

Reply:
This predates me, but I'm pretty sure this is a convenience because a `[]string` of subnet IDs is very helpful for some of the methods attached to the EKS cluster provider struct.

Reply:
We cache the SubnetIDs because I believe they're required to create nodepools. Consider it part of the "base node pool template" which will be used to create any new node pools. (ie: The new node pools will use the same subnets).